qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -113,9 +115,18 @@ def run(self, dag):
                     or ((self.basis_gates is not None) and outside_basis)
                     or ((self.target is not None) and outside_basis)
                 ):
-                    dag.replace_block_with_op(block, unitary, block_index_map, cycle_check=False)
+                    identity = np.eye(2**unitary.num_qubits)
+                    if np.allclose(identity, unitary.to_matrix()):
```

Comment:
`allclose` has an implicit tolerance built-in, so we're potentially removing things that aren't _precisely_ the identity here.  That's sensible, of course, but maybe it's worth making sure the way we handle the tolerance here is consistent with how we treat `approximation_degree` (especially since that's caused issues in the past) and anywhere else that might skip small values (`CommutationAnalysis`, maybe?).

Reply:
Well I figured it was ok like this because there is a similar tolerance built-in to the synthesis routines that normally do this optimization that isn't really adjustable either. The `approximation_degree` usage for synthesis is used for the fidelity of the approximation of the synthesized circuit. I can try to wire the tolerance here in with the value for `approximation_degree`. I guess I'll just have to figure out what values make sense for a mapping of approximation degree which is from `0` to `1` to the tolerance parameters on `allclose`.

Reply:
If an implicit tolerance is something we do elsewhere, I'm happy enough to leave it in place here - I was just concerned with making sure the type of behaviour we have is consistent with itself, whatever it is.

Reply:
That makes sense, just for some references what I was talking about was:

- https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py#L114
- https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/quantum_info/synthesis/two_qubit_decompose.py has a bunch of `isclose()` and `allclose()` calls. The tolerance is adjustable for some (but not all) and where it is adjustable that tolerance value is decoupled from the `approximation_degree` value, which is use to set the `basis_fidelity` argument when the actual decomposer is called.